test/commands/user/set_github_username_test.rb

```diff
@@ -0,0 +1,26 @@
+require "test_helper"
+
+class User::SetGithubUsernameTest < ActiveSupport::TestCase
+  test "sets username" do
+    username = "user22"
+    user = create :user
+    User::SetGithubUsername.(user, username)
+    assert_equal username, user.github_username
+  end
+
+  test "copes with duplicate gh username but unauthed" do
+    username = "user22"
+    user = create :user
+    create :user, github_username: username
+
+    User::SetGithubUsername.(user, username)
+    assert_nil user.reload.github_username
+  end
+
+  test "recalculate pull request reputation for uid matches that change the github_username" do
```

Comment:
Do we maybe also want to test the inverse? That no rep job is created when the github_username stays the same?